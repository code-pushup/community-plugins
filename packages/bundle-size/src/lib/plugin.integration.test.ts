import {describe, expect, it} from "vitest";
import {pluginReportSchema} from "@code-pushup/models";
import bundleSizePlugin from "./plugin";
import {executePlugin} from "@code-pushup/core";
import {omitVariablePluginData} from "testing-utils";

describe('bundleSizePlugin', () => {

    it('should execute correctly and return valid audit outputs', async () => {
       const pluginReport =  await executePlugin(bundleSizePlugin())
        expect(() => pluginReportSchema.parse(pluginReport)).not.toThrow();
        expect(omitVariablePluginData(pluginReport)).toMatchSnapshot();
    });

});

