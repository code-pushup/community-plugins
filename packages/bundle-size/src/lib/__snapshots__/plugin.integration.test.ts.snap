// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bundleSizePlugin > should execute correctly and return valid audit outputs 1`] = `
{
  "audits": [
    {
      "description": "Bundle size audit",
      "slug": "bundle-size",
      "title": "Bundle size",
    },
  ],
  "description": "Bundle size plugin for Code PushUp",
  "icon": "folder-css",
  "slug": "bundle-size",
  "title": "Bundle size",
}
`;
